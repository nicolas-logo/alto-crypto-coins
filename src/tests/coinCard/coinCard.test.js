import React from 'react';
import "@testing-library/jest-dom/extend-expect";
import { render, screen, cleanup } from '@testing-library/react';
import renderer  from 'react-test-renderer';
import CoinCard from '../../components/coinCard/coinCard';
import coinCardMock from '../mocks/coinCardMock';

afterEach(() => {
    cleanup();
})

describe('Coin Card tests', () => {
    it('should render coinCard component', () => {
        render(<CoinCard {...coinCardMock} />);
        const coinCardElement = screen.getByTestId('coin-card');
        expect(coinCardElement).toHaveTextContent('coin mock name');
        expect(coinCardElement).toHaveTextContent('coin symbol');
        expect(coinCardElement).toContainHTML('coin symbol');
        expect(coinCardElement).toContainHTML('https://test.com');
    })
    
    it('should match the snapshot', () => {
        const tree = renderer.create((<CoinCard {...coinCardMock} />)).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should set correctly HTML', () => {
        const instance = renderer.create((<CoinCard {...coinCardMock} />)).root;
        // eslint-disable-next-line testing-library/await-async-query, testing-library/no-node-access
        expect(instance.findByProps({ name: "name" }).children).toEqual([coinCardMock.name]);
    })
    
})