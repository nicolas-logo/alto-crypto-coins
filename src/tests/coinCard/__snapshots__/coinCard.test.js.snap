// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Card tests should match the snapshot 1`] = `
<div
  className="coin-card"
  data-testid="coin-card"
>
  <div>
    <img
      alt="coin"
      className="coin-thumb mx-auto d-block mt-2 rounded-circle"
      height={35}
      src="https://test.com"
      width={35}
    />
    <p
      name="name"
    >
      coin mock name
    </p>
    <p
      name="symbol"
    >
      <b>
        coin symbol
      </b>
    </p>
  </div>
</div>
`;
